### ISSUE TITLE
Allow showing errors to readers

### ISSUE THREAD

**Issue by kellertuer** [rikhuijzer/PlutoStaticHTML.jl#148]
I am currently starting with tutorials for Manifolds.jl using Pluto and PlutoStaticHTML.

In the notebook (preview) at

https://juliamanifolds.github.io/Manifolds.jl/previews/PR534/tutorials/getstarted.html

I have a few cells where I would like to actually have an error to illustrate what error to expect.
Maybe this is also something for Pluto to have a way to nicely expect/display errors, but

For example the cell before 3. The Sphere is quite tricky just to display an error. Could this maybe be made nicer?

Also rendered markdown code and input cells look exactly the same, is there a way to change their appearance?

**Comment by rikhuijzer** [comment]
> For example the cell before 3. The Sphere is quite tricky just to display an error. Could this maybe be made nicer?

Ah yes good question. See https://github.com/rikhuijzer/PlutoStaticHTML.jl/issues/73. If you have the time, then a PR to update the docs here at PlutoStaticHTML could be nice.

Let's leave this issue open until one of us updates the docs

**Comment by kellertuer** [comment]
Maybe this would even be something for PlutoUI – something to nicely display an expected error (and maybe even really throw an error if there is no error to display). Then PlutoStaticHTML could just render that PlutoUI-element “reasonably”?

I find my current solution a little hand-crafted and the error display also does not really look nice.

**Comment by kellertuer** [comment]
I still have to check how this renders in PlutoStaticHMML but the solution I ended up with (adapted from the linked Pluto issue) is

```Julia
macro expect_error(code, error=:DomainError)
	quote
		try
			$(esc(code))
+                       error(string("Error of type ", $(esc(error)), " was not thrown."))
		catch e
			if e isa $(esc(error))
				pretty_error(e)
			else
				rethrow()
			end
		end
	end
end;
```

together with

```Julia
function pretty_error(err) 
    Markdown.parse("""
!!! info "This is how the Error we expect here looks like"
    ```
    $(replace(sprint(showerror, err), "\n" => "\n        "))
    ```

""")
end;
```

**Comment by rikhuijzer** [comment]
That's a bug. I'll see whether I can write a test and patch it. Probably done before tomorrow 12:00

**Comment by kellertuer** [comment]
Cool (because I would have had no idea where to start).

Don't worry its not urgent, its already nice that we found a way to display errors that fast. I have to write quite some part of the tutorial anyways, still. So if it takes longer – no problem :)

**Comment by kellertuer** [comment]
Sorry to bother again – in general the result looks great!

<img width="826" alt="Screenshot 2022-10-13 at 08 48 59" src="https://user-images.githubusercontent.com/115566/195522694-974f3689-3475-4750-8cc0-92a6134f61b5.png">

I just noticed the `e>` (which also only appears for one of the two displays, the other one is fine). I looked at the code and its


```html
  <div class="admonition-body">
    <pre><code>DomainError with 0.1:
    The vector &#91;0.1, 1.0, 1.0&#93; is not a tangent vector to &#91;1, 0, 0&#93; on Sphere&#40;2, ℝ&#41;, since it is not orthogonal in the embedding.</code></pr
  </div>e>
```
( did not change any of the line breaks.

Any idea how this might have happened that the pre is so strangely split?

**Comment by rikhuijzer** [comment]
> Any idea how this might have happened that the pre is so strangely split?

Yeah because I'm trying to parse HTML with basic string operations and regexes which are not a good idea, see https://stackoverflow.com/a/1732454/5056635.

I should have switched to a XML parser 😅

**Comment by kellertuer** [comment]
Then – just maybe – don't regexp 😃

Ok, but then I know where that's from and that its not me ;)

**Comment by rikhuijzer** [comment]
> Then – just maybe – don't regexp :smiley:

Very helpful. Thanks :+1:

**Comment by rikhuijzer** [comment]
> Sorry to bother again – in general the result looks great!
> 
> <img alt="Screenshot 2022-10-13 at 08 48 59" width="826" src="https://user-images.githubusercontent.com/115566/195522694-974f3689-3475-4750-8cc0-92a6134f61b5.png">
> 
> I just noticed the `e>` (which also only appears for one of the two displays, the other one is fine). I looked at the code and its
> 
> ```
>   <div class="admonition-body">
>     <pre><code>DomainError with 0.1:
>     The vector &#91;0.1, 1.0, 1.0&#93; is not a tangent vector to &#91;1, 0, 0&#93; on Sphere&#40;2, ℝ&#41;, since it is not orthogonal in the embedding.</code></pr
>   </div>e>
> ```
> 
> ( did not change any of the line breaks.
> 
> Any idea how this might have happened that the pre is so strangely split?

@kellertuer Is the issue fixed now in v6.0.4?

**Comment by kellertuer** [comment]
yes, it looks perfect locally :) Thanks!
